test/Generics/unbound.swift

```diff
@@ -137,3 +137,19 @@ do {
   assertExactType(of: x1, is: Generic<Int, Bool>.self)
   assertExactType(of: x2, is: Generic<Int, Bool>.self)
 }
+
+// https://github.com/apple/swift/issues/60922
+enum T<E> {
+  // expected-note@-1{{generic type 'T' declared here}}
+  case a(E)
+  case b([E])
+}
+
+extension T? {}
+// expected-error@-1{{reference to generic type 'T' requires arguments in <...>}}
+
+struct G<A, B>: P {}
+// expected-note@-1{{generic type 'G' declared here}}
+
+extension G? {}
```

Comment:
Could you also add an extension of the desugared version of either of these types?